### PR TITLE
content(balance): Changing the lifespan on Hai Trackers to 475 after confirming that they will still do a second pass at over 5.5k range.

### PR DESCRIPTION
**Pull request by Zitchas**
**Bugfix:** This PR addresses issue reported here

## Problem
Hai Trackers, in addition to their many positive qualities, also have sufficient range to hit targets from far beyond the edge of the RADAR/minimap. (technically, at almost double the radius of the radar).

Gameplay perspective:
- Hai Trackers have the longest range of any purchasable weapon in the game.
- It isn't fun to fight against tracking shots that come at you from beyond the edge of the screen and relentlessly home in on you.
- It also isn't fun when you head towards a ship from a long ways away to find out that it started shooting at you from 8400, and by the time you reach 6k distance, it's already got a whole swarm of shots homing in on you. 
- The main other weapon these are comparable to is the AMC, and, unlike the AMC, these have tracking. So all the frustration of being hit from 3k off the edge of the minimap, with the added frustration of it being quite hard to dodge, and even fairly hard to outrun. (unlike torpedoes, which are quite long lasting but quite slow and thus fairly easy to dodge.)

Lore perspective:
- Given the Hai's demonstrated technological capability, it seems odd that they have a weapon whose guidance system, speed, and range rival or exceed the best available from civilizations vastly more powerful than them.
- Given the scale and scope of the Hai conflicts, having a weapon whose range is almost enough to exceed the "fence" seems out of place.

Balance:
- I don't think this should significantly affect the balance of combat, but it may make it slightly less frustrating.

## Fix Details
This changes the lifetime to 475 after significant testing to confirm that the Tracker continues to demonstrate it's characteristic "second pass" capability reliably against targets up to 5.5k from the launcher.

## Testing Done
Fought against ships that use Hai Trackers.

## Save File
This save file can be used to test what it is like fighting against three ships with Hai Trackers. (A Bactrian and two leviathans at the end of the Deep campaign)

[Rebbia Rampart~3023-07-01 Hunting the Bactrian.txt](https://github.com/endless-sky/endless-sky/files/10297360/Rebbia.Rampart.3023-07-01.Hunting.the.Bactrian.txt)

### CODE REVIEW

**Comment by MasterOfGrey** [comment]
One of the key features of a tracker was that if it missed at most normal ranges, it had enough life to circle around and make a second go. 
If this is no longer the case then I am against this change.

**Comment by Amazinite** [comment]
Might be better to instead give Trackers a range override that reduces the range at which they're used while keeping the ability to miss and wrap back around.

**Comment by Zitchas** [comment]
I can check that, however I suspect it still can loop around - 6650 is still a truly massive range, one that really dwarfs or combat engagement ranges. Which is kind of weird, actually, combat is basically point blank only, but then we have this one weapon that is of the radar.

**Comment by EjoThims** [comment]
Range override seems best option to me as well. Set it down to like 5200 or something, even.

**Comment by Hurleveur** [comment]
Yeah range override sounds better.
I was intending for the unfet to force he arfectas into cloaking with trackers and changing this does make it harder given how much it moves around.
Range override would make less shots be wasted and less useless carnage in Wah Ki too.

**Comment by MasterOfGrey** [comment]
> I can check that, however I suspect it still can loop around - 6650 is still a truly massive range, one that really dwarfs or combat engagement ranges. 

I find most reasonably mobile opponents can dodge and then move far enough to negate the circling around with the current range, so I suspect shortening it would worsen it substantially.
A range overide feels better to me.

File: data/hai/hai outfits.txt [hunk 1]
```diff
@@ -306,7 +306,7 @@ outfit "Hai Tracker Pod"
 		"die effect" "tracker impact"
 		"inaccuracy" 20
 		"velocity" 14
-		"lifetime" 600
+		"lifetime" 475
```
Comment (Hurleveur) [inline data/hai/hai outfits.txt:309]:
```suggestion
		"lifetime" 600
		"range override" 5500
```
it takes about 3000 meters for the tracker to turn around and reach the target back in my tests

**Review by Hurleveur — APPROVED** [review]
Less ammo wasting and overkilling sounds good for me!
With this range it'll always be able to hit the target and wont harrass as much at long ranges

**Comment by Zitchas** [comment]
I've made the change, but I have my doubts as to whether or not it is actually going to be a beneficial change. Using the range override like this is effectively giving anything that uses the trackers a huge "anti-kiting" buff, since this will now make it impossible to kite away the AI's ammo from the edge of its actual range.

**Comment by MasterOfGrey** [comment]
It is implicitly the case that your turning radius cannot be your most significant weakness if your lifetime is too short to circumscribe a full loop after a miss.

Also trackers can be jammed (to a degree), so dodging via thrust isn’t the whole story.

**Comment by Zitchas** [comment]
The ship in the save file has both a Large Radar Jammer and a small Radar Jammer, and that seems to have a negligible impact on tracking. So Jamming has to be *much* better for that to be reasonable.

Well, does anyone actually know how much range it takes to do a full loop?

**Comment by Zitchas** [comment]
> It is implicitly the case that your turning radius cannot be your most significant weakness if your lifetime is too short to circumscribe a full loop after a miss.

I disagree. If it's almost always going to have the range to do a full loop and take another run at the target, that's not a disadvantage, that's an opportunity to pile up even more simultaneous munitions that the target will have to deal with simultaneously. It's only a disadvantage if that long turn around time has the result in a significant chance of it *not hitting at all*.

**Comment by warp-core** [comment]
Trackers have optical and infrared tracking. There is no possibility of causing them to lose a lock via radar jamming because they never have a radar lock to begin with.

Small ships should be harder for them to maintain a lock on, and to a greater extent, cooler ships.

**Comment by Hurleveur** [comment]
I think that we need to overall jammers and trackers (and give a jammer useful against these to the Unfettered at some point too) but that's a separate issue, your issue of the trackers shooting at you from a way too absurd range should be fixed now.

**Comment by Zitchas** [comment]
Just doing some testing. With the 475 lifespan I had on them earlier, I can park myself at 4100 distance from a stationary target ship, then point myself in the complete opposite direction from said target ship, and unleash a barrage of trackers and most of them will hit. So they're capable of doing a full 180 arc and still have the lifespan to reach a target 4100 distance away. Which, by the way, is pretty much on the very edge of my radar.

**Comment by Hurleveur** [comment]
I assumed that what was meant was missing the target at a reasonable speed and then doing a full turn, but starting from 0 speed yeah only half that distance is needed.

**Comment by warp-core** [comment]
Trackers have excellent homing, so they should stop accelerating when looking to turn around. So, if the target is stationary and the missile passes it at full speed (somehow), it will need to turn through 180 degrees. With a turn rate of 1.2, this will take 150 frames (2.5 seconds, 1/4 its total lifetime). In that time, with a velocity of 14, it will travel 2100 units.

I'm going to need to check the code for exact acceleration behaviour because the drag makes it more complex, but here's a naive estimate: the acceleration is .7, so it'll take ~40 frames to go from -14 to +14 velocity (relative to its new facing). This process should leave it at the same location it was when it started accelerating again (but drag means it may not be). So, it'll need to traverse 2100 units again (another 150 frames) to reach its target.

Overall, if it misses, it is likely to need a large proportion of its lifetime left to be able to make it back around, and this is assuming a stationary target, and that the lock is maintained throughout the long process of turning around.

**Comment by warp-core** [comment]
Actually, that's not quite right. The missiles should begin accelerating as soon as they've turned through 90°.

**Comment by Zitchas** [comment]
I assume so to. However, it's hard to do carefully reproducible testing with acitvely mobile threats.

So, further testing with a measuring ship, with the above situation of shooting a target that is in the opposite direction, with a lifespan of 475 (and I took the AM off the target ship), and at 4100 range almost all shots hit, with the *diameter* of their 180 turn to face the target is about 1200. 

The whole "They should be able to do a full loop around and hit their target" is nice, but there's a really important detail we're mising here: At what distance should a ship be able to shoot another ship, miss, and still expect it to do a full loop? 1000? 2000? 4000? Right now in vanilla, although I haven't tested it, it feels like it can do a full loop for a second round on a target that is about 6000 from the firing ship. That's a barely reasonable range to begin with, let alone one at which we expect the tracker to take *two* runs at a target at.

Taking two runs at a target, or even three runs at a target, when dealing with an opponent at typical ES combat ranges (0-800, typically) is good. A nice ability for them. Beyond 2k firing distance maybe *a* second pass is ok. But at 6k range... That's beyond combat range. That's "we can't even see it, and only a select very few of our longest range things can even think about targeting stuff out there" range. Having trackers get two passes at a target when fired from 6k+ is pretty unreasonable.

**Comment by Hurleveur** [comment]
In term of context the only good things the Hai have are the Shield Beetles and the tracker pods. The shield regenerator suck, the engines are ok (compared to atomics) and all the rest is nothing special. I don't see an issue with a faction having a few things that got perfected over the (thousands of) years and thus got very good. I don't see a reason why missiles would just cease to exist in space after a given amount of time other than performances anyway.

So yes they are slightly above what we could expect out of them but if the issue is how far they start shooting then that can be fixed by range override.

**Comment by Hurleveur** [comment]
I think we should rebalance all missiles and jammers at some point anyway so it's prob best to leave that for then.

**Comment by Zitchas** [comment]
Just to be clear, I don't want to nerf trackers in terms of overal combat utility, I just want to a) eliminate the "let's swarm the player from beyond radar range" effect, and b) do so while retaining the ability to kite them (which means that absolute range needs to be at least in the same ballpark as the override range. So far, my tests show the range override, while reducing a, also makes them *more* effective and much harder to kite, so not really effective in improving the player experience.

I'm perfectly fine with them remaining highly effective and capable of circling around (especially in CQC.) I'm perfectly fine with trackers being able to take 4 runs at a target if fired at point blank range.

**Comment by Hurleveur** [comment]
I dont think all missiles should be avoidable just by running away

**Comment by Zitchas** [comment]
Well, the other problem here is that we have two problems here. I'd prefer to focus on the weapon itself. The usage is something that, honestly, the AI should handle, not have hard-baked into the weapon.

For instance, most ships should just start shooting as soon as their target is in range. A canny veteran ship might hold fire until 4k and the target won't be able to get away. A brawler might not use them until under 1k range and basically use them like a Pug seeker - aka a super long range homing gun.

Any application of range override is basically forcing *every* ship that uses a particular weapon into a single specific method of using a weapon.

Anyway, back to the real question I asked earlier: If the "swinging around for a second run at a target" is essential behavior, what is the maximum practical range that it should be able to take a second run at a target? 

Keeping in mind that about 99% of ES combat takes place at <600 range. Do we really want or even need to have something that is capable of annihilating targets predictably and consistently at 8k? This sounds like it should be a weapon whose optimum (with a second pass) range should be around 2-4k, which is far outside most ship's engagement zones, but crucially *within the radar sight range* of the player. That's really the sticking point here, I think. We're not talking about some super rare weapon or some big slow thing like a torpedo. This is a dirt common weapon, easily accessible, and quite a fast one. If it was relatively rare and unlikely for the player to come up against it, it'd be another matter. But fairly new players shouldn't be getting hit by fast, quite accurate, and reasonably hard hitting missile that dodges asteroids and comes from over two radar screen away. (missile description aside, this thing has pretty good tracking. Yeah, it might take it a long time to circle around, but that's only assuming it misses in the first place. I honestly don't see it miss very often.)

**Comment by Hurleveur** [comment]
Range override is treated as range, when we get proper personalities in we could change the behavior.
If you just want to give it range override than change can happen quickly we all agree on it, and would resolve the problem announced by this PR : shooting from too far, not being too strong.

This weapon is the apogy of Hai tech, I dont think it should be nerfed without considering all the impact that can have, not nessecarely on gameplay but also on lore. I'd like to for instance see it nerfed and the shielding buffed, and maybe even other changes.

**Comment by Zitchas** [comment]
I still haven't heard anyone say what they actually think it's range should be. At what distance should one expect that a shot will be able to miss and complete a circuit and take a second run at a target?

And what all this is really driving home is that we need a better way to handle long range combat, because having things shoot at us from beyond radar range just plain isn't fun.

**Comment by EjoThims** [comment]
https://github.com/EjoThims/Weapon-Ranges

I really think the solution is -close- to being this simple.

I do need to update that with my more recent local changes, though. 🤔

I myself have always thought tracker range was good as is, but I still see a range override here as the best way to ensure the desired behavior of cycling (even repeatedly) on a miss. And I do still see that -as- the desired behavior. If that override needs to be lowered further, then so be it.

Perhaps what we really need for this situation is a new attribute, something like "lock range," at which the projectile loses its target tracking? Drop tracker "range override" to like 3.5k and then give them a 4k "lock range"?

Firestorms could use that as well... double their lifetime to really drive home the "slow but persistent" homing feel, but let them be outdistanced instead of only outlasted.

**Comment by Hurleveur** [comment]
If we're changing trackers what about changing their hit force too? It's a bit too hai if you see what I mean
I think we should open a PR to rebalance all of Hai outfits

**Comment by Zitchas** [comment]
> Perhaps what we really need for this situation is a new attribute, something like "lock range," at which the projectile loses its target tracking? Drop tracker "range override" to like 3.5k and then give them a 4k "lock range"?

I'd love to see a "lock range" that would basically represent the projectile using the launching ship's targeting system, and as such loses its lock if the target exceeds a certain distance from the launcher. (and the AI automatically using the lock range if it is shorter than the total range to decide when to fire). Having the Tracker have a "lock range" of about 4k would fully address my concerns, while allowing the tracker to retain it's current lifespan.

**Comment by Zitchas** [comment]
Alrigh, I just did some practical testing.

Testing parameters:
- Tracker lifespan set to 475
- No range override
- Launching ship is stationary, and has just a single tracker pod, so I have the best possible chance to focus on each tracker shot individually to try to dodge it.
- I'm flying an Osprey with a max speed of 636, an acceleration of 305, turning 138, and volcano afterburners. (I think this is a fairly good representation of a fast warship. I know it's not *the fastest*, but it's definitely on the high side)
- My ship idles at negative heat, although it builds up to 14-25% under constant manouvering.
- My ship has 1 bullfrog antimissile
- I'm just juking back and forth trying to dodge trackers and stay at a relatively consistent distance from target.

Results: (the distances are +/- 200, as I can't dodge 100% perfectly perpendicular to the flight path all the time)
4k distance:
- Most trackers hit. A small number (<10%) seem to hit asteroids or lose their lock. A few are shot down (<20%) I manage to dodge about 15%. As in I manage to dodge about 1 tracker out of every six that reaches me.
- Of the trackers that I do manage to dodge, in excess of 95% of them complete their turn around and make a second pass at me. Some even manage a third pass.

5k distance:
- Pretty much the same as for 4k, although the number that hit asteroids or lose their lock increases to about 15-20%, but I seem to do poorer at dodging. I probably only dodge about 1 in 7 that reach me.
- Of the trackers that I dodge, it gets interesting. Depending on which way I was dodging relative to the tracker path, *sometimes* it fizzles out before getting a second pass, but most of the time it still manages to get a second pass. It feels like if I dodge at the perfect angle, it probably won't get a second pass, but it happens rare enough that it *could* just be losing the lock.

5.5k distance:
- This seems to be about the limit of the second pass distance. At this point, trackers that I dodge usually don't get a second pass, although if I dodge poorly they still sometimes manage to get one.

6k distance: 
- Trackers never get a second pass, but I do note that they still have enough lifespan that some trackers that lost the target lock can still regain the lock after having veered significantly off course and still reach me. (and, for lols, some that I sucessfully dodge do still hit me when I dodge the perfectly wrong direction and they don't have to do a full loop to get back on target. It's rare, but it did happen)

2k distance:
- The very few trackers that miss always get a second pass, the miniscule quantity that I actually manage to dodge twice get a third pass >50% of the time.

-----

Based on this, I feel that the 475 lifespan results in the trackers retaining their characteristic "second pass capability" in an unreduced capacity up to 5.2k distance, with gradually reducing likelyhood of being able to effectuate a second pass above that until they basically lose the ability to do so entirely at around 6k.

As such, this significantly reduces the problem described in my initial post, and brings the effective "engagement zone" of the trackers within the scope of the radar. While they can still reach beyond this (they can still hit up to 6.6k vs the radar's edge at about 4.5k), the reduction in range prevents them from building up massive swarms as one closes from 9k+ distance to the edge of radar contact range. This makes the combat feel more personal, as one actually has the chance to catch a glimpse of one's opponent, and means that if one dies to them, one will at least have the possibility to see (on the radar, anyway) the person who killed you.

Note: This change actually makes the Tracker a bit *more* effective, since that's potentially 3k less distance over wich to lose a lock, run into something else, etc. So this makes it slightly more efficient on an individual basis.

**Comment by Hurleveur** [comment]
Thats a long title alr
I still dont think we should nerf these without buffing another thing at the same time.

**Review by Hurleveur — COMMENTED** [review]
Just removing my approval, still not set about these changes (not like it matters that much but still)

**Comment by Zitchas** [comment]
This isn't the place for discussion about artillery, although I would like to see artillery in ES.

Trackers aren't artillery. Artillery, in my books, is strictly unguided munitions. See railguns in EVN for a good example of effective artillery in this style of game.

I must admit, I am slightly disappointed. There were lots of complaints about nerfing a key Hai ability, and I have repeatedly asked for clarification about what distances that ability needs to work at, with no responses whatsoever. I have demonstrated that over the entire field of view of our radar this key capability will be maintained, and the only response is that "some other Hai thing needs to be buffed to compensate" and a statement that we should have *more* over the horizon death, still with no explanation as to how having more streams of unavoidable death from over the horizon is even remotely beneficial.

**Comment by Hurleveur** [comment]
Range override is the easy way to get this trough tbf
I have no strong opinions on this change if it doesnt do that I just dont approve or disapprove it is all
I dont ever use missiles anyway, but reducing range would make anti arfie attack harder (though I guess bombs could just be used too hence why I'm neutral on it)

**Comment by Hurleveur** [comment]
Sorry I was just hoping I could get rid of that approval but github is stupid.

**Comment by Zitchas** [comment]
@Hurleveur 

You wanted a compensating buff to some other Hai outfit? Alright, how about this? Buffing the Hail Williwaw cooler up to a value of 12, which shifts it from being slightly worse than water cooling to being slightly  better than liquid nitrogen. The human set of coolers will still be better for massive heat loads with the Liquid  Helium, but the Hai one will be better for medium sized heat loads and definitely won't be a joke anymore.

**Review by Hurleveur — APPROVED** [review]
Sounds good to me for sure, that cooling being a joke for so long was weird.

It makes sense that they wouldnt need bigger heat managers but they fought dragons and I'm pretty sure those used a lot of heat so Hai had to adapt. Either that or for an eventual conflict with humanity which they would have prepared for.
Also their ships running a tad bit cooler helps to explain why they lasted so long. And their shields do heat them a bunch.

As for the tracker its true it always was a bit ridiculous, to the point a relatively small mass of ships with these (15 * 9 of thess to be exact) forces 2 arfectas to cloak with ease. This should make it a bit harder and force smth like a surround.

**Comment by Zitchas** [comment]
@MasterOfGrey 
> One of the key features of a tracker was that if it missed at most normal ranges, it had enough life to circle around and make a second go.
> If this is no longer the case then I am against this change.

No one has provided any information on what they consider to be "normal ranges," and typical combat ranges are <800; so I'm assuming 1k is "normal ranges," and the Tracker's exceptional / special normal engagement distance is everything within the scope of the radar. (roughly 4500 radius) As such, my testing demonstrates that the tracker maintains its full "second pass" capabilities up to roughly 700 *beyond* the edge of the radar, and some second pass capabilities up to 1k past the edge of the radar. (5200 and 5500 respectively) At below 4k it is even capable of sometimes effectuating a 3rd pass, but my dodging is sufficiently poor that I can't reliably test third pass capabilities. Is this sufficient to meet your criteria of being able to circle around to make a second pass "at most normal ranges?"

Additionally, Hurleveur has been particularly interested in seeing something else buffed as "compensation" for this change. As the Hai lead, are you comfortable with / liking the change to the Williwaw cooler being buffed in this way? It still isn't better than the largest human coolers, but it will now at least be an appealing option for cooling on small and medium ships.

**Comment by MasterOfGrey** [comment]
I get preoccupied for 2 days and ya'll have a thesis level debate in here. Geez.

Couple things I would like to address as asides (just for clarity):
>within the radar sight range of the player.

If we're referring to the little radar display (not just your screen when you zoom out), I've repeatedly advocated for the distance displayed on that to fall off logarithmically so that much more distance is shown directly rather than just as icons on the outer circle. I'm not sure if that helps here though.

> If it's almost always going to have the range to do a full loop and take another run at the target, that's not a disadvantage, that's an opportunity to pile up even more simultaneous munitions that the target will have to deal with simultaneously. It's only a disadvantage if that long turn around time has the result in a significant chance of it not hitting at all.

Linguistically, and tactically, from a single weapon perspective:
- If you fire and miss your shot, that's a potential disadvantage if your weapon's behaviour/accuracy makes that more likely. (and any long-range weapon with non-trivial flight time to contact falls under that category)
- - but if it can come around again - then that's an advantage. 
- If it takes a long time though, then *you're taking fire in the meantime* - that's a *dis*advantage.
- - If the *only* disadvantage it has is that time during which you take fire while hoping it returns (because it takes a long time), then it *must* be capable of making that return.

That's how that works. That's what that means.

Alright, setting that aside:
## Onto the Actual Point.
>Testing parameters:

This looks quite thorough, and I would say I am satisfied that they retain their essential characteristic in this case.

>No one has provided any information on what they consider to be "normal ranges," and typical combat ranges are <800; so I'm assuming 1k is "normal ranges,

It has been christmas time and people aren't exactly the most blessed with responsiveness, but this set of assumptions feels suitable based on my experience and on reading through the thesis discussion up there.

>As the Hai lead, are you comfortable with / liking the change to the Williwaw cooler being buffed in this way?

Honestly, I don't particularly care for the Hai cooler, and I see the fact that it's bad to be something of a deliberate decision that could justifiably have been made in the past.
When you were cramming Hai into a small area and trying to suppress dissent, it would make sense to me that removing better coolers from the market would make it harder for a rebellious faction to cram a ship full of high-performance combat outfits and start causing trouble.
To my mind, not having a better cooler available (given how long they've been around) feels like a political decision cloaked in a long history of economic or sustainability based "justifications".
- "No we can't use liquid nitrogen cooling, it wastes it detracts from our biosphere maintenance."
- "No we can't use liquid helium, it's a reserved scientific resource and it wouldn't be economic to price it for both uses."
- etc. etc.
"Williwaw" refers to a squall blowing offshore from a rough coast. To me this sounds like some sort of evaporative air cooling system, and right now it's strictly better than the human water cooling when previously it was marginally worse.
(1.1875 cooling per space vs 1.2 - compared to the human air cooling of 1.)
A cooling value of 12 makes it superior to the liquid nitrogen cooler which, to my mind, is quite unacceptable.
I wouldn't be against increasing the cooling to maybe 10.5, or even 11 at a stretch, which would make it better by weight than the human water cooler, but with a lower absolute value. This would imply that they've refined a relatively inferior technology to be pretty much at the peak of performance, which I find reasonable. 
It would also mean that it has a niche, in that if you really do need the extra cooling the extra 2 tons for the human water cooler might be worth it for the absolute benefit, or if you have a need for *large* cooling then you'd step up, but you could get better bang for buck at the small end with a williwaw.
In the spectrum of human+hai technology which is the most common blend for new players and for ship variants used in the game, this does give it a specific niche it can occupy which is arguably a little better than it's current niche of *just* being an option between cooling ducts and water cooling (which it would retain).

tl:dr maybe 10.5 or 11 as a buff, not more

In the longer term, I would like to see the Hai have some kind of holographic projection-based optical jammer as the more specific counter to Trackers without having the broader heat management implication.

**Comment by Amazinite** [comment]
> Compensating Hai Buff to Williwaw

I don't see how buffing Hai cooling follows as a compensation for nerfing Tracker range. If we increased firing heat then it'd make sense, but this isn't that.

**Comment by Hurleveur** [comment]
Idea lf it mostly is to not just make the Hai worse, if they didnt develop trackers as much they could have done smth else
And what I meant when asking for smth like that was to do a proper rebalance PR although if its just a small buff to the williwaw that miserably sucks that'd be fine here

**Review by EjoThims — CHANGES_REQUESTED** [review]
Going to second a more in depth balance pass at Hai tech at a later date; I honestly feel it's pretty much mandatory to ensure smooth integration of plans to expand their arsenal anyway.

I'm also going to second accepting this as a stopgap in the meantime with the Walliwaw bumped to 11 but not 12 (though reviewing/refining that further should be part of the above still).

And just as an aside for clarity, my proposal for lock distance was to have it be the distance between the target ship and the projectile itself, but getting into that is definitely for a separate issue/PR. :)

**Comment by warp-core** [comment]
Something interesting: Trackers don't actually have the greatest range of any purchasable weapon (ignoring licenses). Thunderhead missiles can actually achieve a greater range. The game lists a range of 3200 based on the product of the initial velocity (4) and the total lifetime (400+400). But, both the first stage and the submunitions can accelerate up to maximum velocities of 11 (if the first stage was already at, or close to, this speed, the submunitions will actually be spawned with higher speeds and then dragged down to 11). While this is still lower than the Tracker's speed of 14, the greater lifetime (800 total) means you can hit a target with Thunderheads from slightly further away than it can hit you with Trackers (I have tested this).

This isn't something you will have seen in game unless you've decided to fight the Wanderers for some reason (I suspect most of us wouldn't bother with this) and, even then, since the game calculated the range to be 3200, they won't fire them at you from beyond that range, you'd have to get within that range and then let the missiles pursue you along a path that traverses 8k units. Though, if you get close enough to the first stage, they may split early, reducing the lifetime by however long the first stage had left, and potentially reducing the velocity that the submunitions start with.

**Comment by Zitchas** [comment]
That's interesting to know. But you are correct: I haven't fought the Wanderers, and everything deathballs, so I don't have any experience with thunderheads reaching anywhere close to that kind of range. It does, however, seem like a very long way.

**Review by EjoThims — APPROVED** [review]

**Review by MasterOfGrey — APPROVED** [review]
